src/js/components/dialog/Dialog.jsx

```diff
@@ -0,0 +1,212 @@
+// @flow
+import React, {Component, PropTypes} from 'react';
+
+import {ModalContainer} from './ModalContainer';
+
+//--------------------------------------------------------------------------
+//
+//  Basic Dialog
+//
+//--------------------------------------------------------------------------
+
+/**
+ Basic, bare-bones Dialog component:
+
+ - Show the user a styled dialog, with any child content to be set by owner.
+ - Listen for and re-dispatch user dismiss requests.
+
+ */
+export class BasicDialog extends Component {
+
+    props: {
+        className?: string,
+        onDismiss?: Function,
+        ignoreEscapeKey?: bool,
+        children?: ReactChildren
+    };
+
+    //--------------------------------------
+    //  User Interaction
+    //--------------------------------------
+
+    modalScreenClicked = () => {
+        const { onDismiss } = this.props;
+
+        if (onDismiss) {
+            onDismiss();
+        }
+    };
+
+    keyPressed = (event:KeyboardEvent) => {
+        // TODO: Move this into ModalContainer, and remove from Popover as well!
+        const { onDismiss, ignoreEscapeKey } = this.props;
+
+        if (!ignoreEscapeKey && onDismiss && event.keyCode === 27) {
+            onDismiss();
+        }
+    };
+
+    //--------------------------------------
+    //  React Lifecycle
+    //--------------------------------------
+
+    render() {
+
+        const {className, children} = this.props;
+        const newClassName = (className ? className + " " : "") + ' Dialog';
+
+        return (
+            <ModalContainer onScreenClick={this.modalScreenClicked}>
+                <div className={newClassName}>{children}</div>
+            </ModalContainer>
+        );
+    }
+
+    componentDidMount() {
+        document.addEventListener("keyup", this.keyPressed, false);
+        // TODO: Remove from here and Popover
+    }
+
+    componentWillUnmount() {
+        document.removeEventListener("keyup", this.keyPressed, false);
+        // TODO: Remove from here and Popover
+    }
+}
+
+BasicDialog.propTypes = {
+    onDismiss: PropTypes.func,
+    ignoreEscapeKey: PropTypes.bool,
+    children: PropTypes.node
+};
+
+//--------------------------------------------------------------------------
+//
+//  Dialog Header / Title Bar
+//
+//--------------------------------------------------------------------------
+
+/** Basic header for dialogs */
+export class DialogHeader extends Component {
+    props: {
+        children?: ReactChildren
+    };
+
+    render() {
+        return (
+            <div className="Dialog-header">
+                <h3>{this.props.children}</h3>
```

Comment:
In the event someone wanted to use `Dialog` but provide custom header content (icons, a button, etc), should we do anything a bit fancier here? Perhaps check if `children` is a string and wrap in h3, if a `ReactElement` omit the h3?

I know they can always use `BasicDialog` and built up their own, just wasn't sure it was a use-case we wanted to bother supporting.